### PR TITLE
Added a pixel-detail minimap option to tiles mode.

### PR DESCRIPTION
**Pull request by CIB**
Minimap Example

### CODE REVIEW

**Comment by CIB** [comment]
The problem is that with the narrow sidebar, there's hardly any space there to put a decently sized minimap.

**Comment by stk2008** [comment]
Oh I like that

I always miss the text of where the monsters are this I can see at a glance
and even where I like this a lot :)

**Comment by stk2008** [comment]
Edit

Could we have an option to even zoom in and out of mini map ?

**Comment by CIB** [comment]
Rollin'

**Comment by stk2008** [comment]
This really is cool man :)

**Comment by vache** [comment]
I would suggest using a really high contrast color for creatures in the minimap.  Just looking at that, I cant really tell what was a bee and what was terrain.  Maybe make monsters yellow?

**Comment by stk2008** [comment]
How about a really fast flashing yellow

**Comment by CIB** [comment]
Yeah, they should probably flash or something. But it's apparent enough in the game because creatures tend to move whereas terrain doesn't. There's also the option of replacing the entire "monster distance" UI and somehow directly marking each monster type on the map.

**Comment by Rivet-the-Zombie** [comment]
Oh wow that's really slick!

**Comment by vache** [comment]
Hmm, I'm red-green colorblind and even if the dots were moving I think I would still have a really hard time noticing.  Flashing would probably work though.

**Comment by Rivet-the-Zombie** [comment]
Yeah, flashing would be best. Static yellow dots would be useless to a player with tritanopia like me.

**Comment by Zireael07** [comment]
This looks really cool! And agreed that the dots ought to flash.

**Comment by CIB** [comment]
okay

**Comment by stk2008** [comment]
Niceeeeeee

**Comment by stk2008** [comment]
Ah no this PR failed :(

Was looking forward to this one :)

**Comment by CIB** [comment]
Screenshots of various window sizes and sidebar styles:

[http://i.imgur.com/itV7BlJ.png]()

**Comment by CIB** [comment]
I've been playing with these changes for several days now, and it works fine IMO from a gameplay perspective. Should be ready to merge.

**Comment by stk2008** [comment]
Yeahhhhhhhhhh awesome no more of me being a noob and missing important text about a monster near me.
Even worse I disable it into safe mode and most of the walk into them LOL.

This will be epic a quick glance and bam I know there is monsters and even better where :)

Thanks

File: src/cata_tiles.cpp [hunk 1]
```diff
@@ -887,6 +888,90 @@ void cata_tiles::draw( int destx, int desty, const tripoint &center, int width,
...
+    for( int y = 0; y < tiles_y; y++) {
+        for( int x = 0; x <= tiles_x; x++) {
+            tripoint p(start_x + x, start_y + y, center.z);
+
+            lit_level lighting = ch.visibility_cache[p.x][p.y];
+            if(lighting != LL_DARK && lighting != LL_BLANK) {
+                const auto critter = g->critter_at( p, true );
+                if( critter != nullptr ) {
+                    SDL_Color color = cursesColorToSDL(critter->symbol_color());
+                    for(int xOffset = -tile_size_x; xOffset <= tile_size_x; xOffset++) {
```
Comment (BevapDin) [inline src/cata_tiles.cpp]: It works fine and all. But this loop should probably go into its own function (something like `draw_rhombus` or so) as it's currently a bit difficult to make out what this code does.

File: src/cata_tiles.cpp [hunk 2]
```diff
@@ -887,6 +888,90 @@ void cata_tiles::draw( int destx, int desty, const tripoint &center, int width,
...
+            lit_level lighting = ch.visibility_cache[p.x][p.y];
+            SDL_Color color;
+            if(lighting == LL_DARK || lighting == LL_BLANK) {
+                color.r = 12;
+                color.g = 12;
+                color.b = 12;
+            } else {
+                int veh_part = 0;
+                vehicle *veh = g->m.veh_at( p, veh_part );
+                auto furniture = g->m.furn_at( p );
```
Comment (BevapDin) [inline src/cata_tiles.cpp]: Unused variable. Why does the compiler not warn about this?

File: src/cata_tiles.cpp (hunk 1 above)
Comment (CIB) [inline src/cata_tiles.cpp]: Agreed.

File: src/cata_tiles.cpp [hunk 3]
```diff
@@ -887,6 +888,90 @@ void cata_tiles::draw( int destx, int desty, const tripoint &center, int width,
...
+                if (veh != nullptr) {
+                    color = cursesColorToSDL(veh->part_color(veh_part));
+                } else if ( g->m.has_furn( p ) ) {
+                    auto furniture = g->m.furn_at( p );
```
Comment (BevapDin) [inline src/cata_tiles.cpp]: This (and `terrain` below) should be references (`furn_at` returns references, theres is no need to copy them).

File: src/cata_tiles.cpp (hunk 2 above)
Comment (CIB) [inline src/cata_tiles.cpp]: Good catch. It probably does notice. I need to get an IDE set up that shows the warnings in-editor..

**Comment by CIB** [comment]
Thanks a bunch for the review and feedback!

File: src/cata_tiles.cpp (hunk 2 above)
Comment (narc0tiq) [inline src/cata_tiles.cpp]: PR builds are Lincurse builds, if that's the compiler you were thinking of.

**Comment by Night-Pryanik** [comment]
I really like this function, but it is just too big for the log. Is it possible to draw a smaller variant (as in second screenshot) where the old minimap is located?

**Comment by stk2008** [comment]
im loving this so far to.

I cant see the enemys flashing on the mini map though are they meant to?.

If they aint would it be easy to have them flash and only monsters flash
that way it will be realy easy to tell.

Thanks

**Comment by Robik81** [comment]
Oh man, this one is epic, another game changing addition from you.

**Comment by stk2008** [comment]
Spotted an error when I shift and X to peek mini map does not update to
show this.

**Pull request merged**